### PR TITLE
Add new created subnet to allowed subnets into StorageAccount

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/11936

### CODE REVIEW

File: service/controller/resource/subnet/resource.go [hunk 1]
```diff
@@ -152,11 +153,53 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+				return microerror.Mask(err)
+			}
+
+			storageAccount, err := storageAccountsClient.GetProperties(ctx, key.ClusterID(&cr), key.StorageAccountName(&cr), "")
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if !isSubnetAllowedToStorageAccount(ctx, storageAccount, currentDeployment) {
+				*storageAccount.AccountProperties.NetworkRuleSet.VirtualNetworkRules = append(*storageAccount.AccountProperties.NetworkRuleSet.VirtualNetworkRules, storage.VirtualNetworkRule{VirtualNetworkResourceID: nil})
```
Comment (nprokopic) [inline service/controller/resource/subnet/resource.go:169]: I'm probably missing something, but how does this add the current subnet to the allowed subnets lists?

File: service/controller/azure_cluster_controller.go [hunk 2]
```diff
@@ -119,56 +116,56 @@ func NewAzureCluster(config AzureClusterConfig) (*AzureCluster, error) {
...
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
-		}
-
-		releaseResource, err = release.New(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
+	//var azureClusterConfigResource *azureclusterconfig.Resource
```
Comment (nprokopic) [inline service/controller/azure_cluster_controller.go:119]: Maybe remove this commented code before merging (these things tend to build up), we can always check out git history :)

Comment (fiunchinho) [reply service/controller/azure_cluster_controller.go:119]: it's commented out so I can test :sweat_smile: , I'll undo it now, since it's working fine.

**Comment by fiunchinho** [comment]
Tested it on `godsmack` , it's working fine.

**Pull request merged**